.github/workflows/ci.yml

```diff
@@ -1,55 +1,23 @@
-name: CI
-on:
-  push:
-    branches-ignore:
-      - 'master'
+on: [push, pull_request]
 
 jobs:
-  format:
-    name: Format
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run format check'
-        run: cargo fmt -- --check
-  dependencies:
-    name: Dependencies
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run package check'
-        run: cargo check --all-targets --all-features
-  linter:
-    name: Linter
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run linter check'
-        run: cargo clippy --all-targets --all-features -- -D warnings
-  unittest:
-    name: Unittest
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run unit tests'
-        run: cargo test --all-targets --all-features
+    ci:
+        strategy:
+            matrix:
+                rust: [stable, beta]
+        runs-on: ubuntu-latest
+        steps:
+            - name: Install Rust toolchain
+              uses: actions-rs/toolchain@v1
+              with:
+                  components: rustfmt, clippy
+                  override: true
+                  profile: minimal
+                  toolchain: ${{ matrix.rust }}
+
+            - uses: actions/checkout@v2
+
+            - run: cargo fmt -- --check
+            - run: cargo check --all-targets --all-features
+            - run: cargo clippy --all-targets --all-features -- -D warnings
+            - run: cargo test --all-targets --all-features
```

Comment:
When you put all checks in one job they are not running in parallel, which makes them slower for execution. Also, I wanted to start for all of them from a clean copy of the repository.

Reply:
We can still have this with less configuration